### ISSUE TITLE
google login error 16

### ISSUE THREAD

**Issue by shawns582** [EddyVerbruggen/cordova-plugin-googleplus#529]
I am using this plugin in my Ionic 3 app. It was previously working great, but now it is returning "google login error 16"

Any idea what is causing this or how to fix it? Thanks!

**Comment by TimeTravelersHackedMe** [comment]
On Android:

I'm getting the same error.. I get error code 4 when I try silentLogin and then use that code to run regular login... The Google Sign-In prompt opens up but when I click a user account the function errors out with the code of 16.

**Comment by shawns582** [comment]
Im also experiencing this on Android (havent tried ios yet). I think I am also using play-services 15.0.1 (build.gradle and project.properties use "+" for the version number). I tried changing both build.gradle and project.properties to use 12.0.1 and I still get the same error. I also tried plugin ver 5.3.0 (instead of 5.3.1)..but no luck.

**Comment by TimeTravelersHackedMe** [comment]
What's your plugin list stack? Here's mine:

cordova-android-play-services-gradle-release 1.4.3 "cordova-android-play-services-gradle-release"
cordova-android-support-gradle-release 1.4.4 "cordova-android-support-gradle-release"
cordova-launch-review 3.1.1 "Launch Review"
cordova-plugin-actionsheet 2.3.3 "ActionSheet"
cordova-plugin-appavailability 0.4.2 "AppAvailability"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-facebook4 2.4.0 "Facebook Connect"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-firebase 1.0.5 "Google Firebase Plugin"
cordova-plugin-googleplus 5.3.1 "Google SignIn"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-ionic 5.1.5 "cordova-plugin-ionic"
cordova-plugin-ionic-webview 2.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-media 5.0.2 "Media"
cordova-plugin-music-controls 2.1.4 "MusicControls"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-velda-devicefeedback 0.0.2 "DeviceFeedback"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-plugin-x-socialsharing 5.4.0 "SocialSharing"
cordova-plugin-x-toast 2.7.0 "Toast"
es6-promise-plugin 4.2.2 "Promise"
im.ltdev.cordova.UserAgent 1.0.1 "User-Agent"
ionic-plugin-keyboard 2.2.1 "Keyboard"
twitter-connect-plugin 0.6.0 "Twitter Connect"

**Comment by restivof** [comment]
Hi,

I have the same problem. Are you resolve the error 16 ?

Thank you.

**Comment by JulienSopheos** [comment]
Hi, I've the same error. Tried with several versions (5.3.1/5.3.0/5.2.0/5.1.0/5.0.0)

It was working, then without any code change the problem started occurring.

Not sure it's helpful, but there's the [documentation about CommonStatusCodes](https://developers.google.com/android/reference/com/google/android/gms/common/api/CommonStatusCodes#CANCELED)

> public static final int CANCELED
> The result was canceled either due to client disconnect or cancel().
> 
> Constant Value: 16

**Comment by NikolaBalaban** [comment]
For Ionic 3 use these exact versions:

"@ionic-native/google-plus": "4.7.0"
"cordova-plugin-googleplus": "5.2.1"

And sign your app even in develop mode, like this: https://www.webniraj.com/2016/11/21/cordova-changing-the-default-debug-keystore-for-android-applications/
or test with production build

**Comment by JulienSopheos** [comment]
Thank's for your suggestion, but i still have the same issue.

I tried with your exact versions recommendations, don't work in development neither in production.

**Comment by restivof** [comment]
Thank you Nikola, but don't work.

I created a new blank project. 
And a new project in google console API and in firebase
I install "@ionic-native/google-plus": "4.7.0" and "cordova-plugin-googleplus": "5.2.1"
In Firebase i active google connect.

I try the client_id generate by firebase and the client_id generate manually.

Which stage did I miss ?

My configuration : 
IONIC 3.20.0
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-googleplus 5.2.1 "Google SignIn"
cordova-plugin-ionic-keyboard 2.1.2 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"

Thank you.

**Comment by NikolaBalaban** [comment]
So I had the same problem 3 days ago. Solved it with versions and signing as mentioned in previous comment.
My Cordova CLI version is 8.0.0
Ionic version is 3.20.0
cordova-android (android platform) is 7.1.1

I used Cliend ID generated for **Web Client** in Google Cloud Platform, in credentials section.
I made REVERSED_CLIENT_ID using that Client ID and installed the plugin version 5.2.1.
Like so: 

ionic cordova plugin add cordova-plugin-googleplus@5.2.1 --variable REVERSED_CLIENT_ID="com.googleusercontent.apps.1234567890-abcdefghijklmnopqrstu"

As you can see, I did not add WEB_APPLICATION_CLIENT_ID variable.

And when calling "login" function for google plus plugin i provided webClientId using the same Client ID (but not reversed). Like so:

this.googlePlus.login(
        {'webClientId': '1234567890-abcdefghijklmnopqrstu.apps.googleusercontent.com'}
        ).then(
        googleData => {console.log(googleData)},
        error => {console.log(error)}
        );

Also, try doing "cordova clean" comand when you change plugins, sometimes it helps...

**Comment by restivof** [comment]
thank you for the clarification.
Don't working with the same configuration and code.

which redirection URI for the configuration web client in Google console API did you put ?

**Comment by NikolaBalaban** [comment]
I've placed 3 of them  in Authorised Javascript origins:
http://localhost
http://localhost:8100/
https://myappname-12345.firebaseapp.com

And in authirised redirect URIs:
https://myappname-12345.firebaseapp.com/__/auth/handler

That reminds me. Firebase has "Set up sign-in method". Try enabling Google login there.

Hope it helps =]

**Comment by asoap** [comment]
I just got this working in developer mode.  I'm not exactly sure what I did.  But I got it to work.  Also this is my first time going through this, I haven't had this working in the past. 

The last thing that I did which made it work is setting an Authorized Redirect URL within google's settings page.  I did this under the credentials of "Web client (Auto-created for Google Sign-in)"

Other things.  I used google plus version 5.2.1

I also created a key.  I did NOT follow the steps NikolaBalaban mentioned with setting a custom key store.  (I might have to do that for a release, but I'm not sure) 

I did create a KEY, not sure if that had any impact or not.  Following the instructions here:
https://developers.google.com/drive/android/auth

Because I run windows I had to use this command:

`keytool -exportcert -alias androiddebugkey -keystore C:\Users\[ your user name here ]\.android\debug.keystore -list -v
`
I even followed the SHA1 instructions on that page and created a special credentials for an android app, but that didn't seem to make any effect. 

I lastly went back into the "Web client (Auto-created for Google Sign-in)" and created a redirect URL to my development server.  And suddenly it's working again. 

I hope this helps.

**Comment by camposmichel** [comment]
Hello everyone, I solved using the versions
**"@ ionic-native / google-plus": "4.7.0"
"cordova-plugin-googleplus": "5.2.1"**
And adding the SHA1 debug in my Firebase application:
**keytool -exportcert -alias androiddebugkey -keystore "C: \ Users \ Michel Fields \ .android \ debug.keystore" -list -v**
**The default password is "android".**

It worked for me.

**Comment by A110SHT** [comment]
@camposmichel  thank you it worked for me
cordova v 8.0.0
cordova android version 7.1.1
googleplus login 5.2.1

**Comment by Dan890** [comment]
@camposmichel dosn't work for me :(
I inserted the SHA1 debug in my Firebase app on past -and  it is part of my google services.json
still get error 16 on debug mode.

**Comment by SeaMeow** [comment]
@camposmichel solution worked for me. Had to insert the correct SHA certificate fingerprints in firebase console settings for android.

**Comment by bhutada1992** [comment]
same problem.Made lots of changes but no result.

**Comment by bhutada1992** [comment]
@Dan890  hey can please give steps to solve this issue?

**Comment by restivof** [comment]
It's WORK ! ! !
Thank you to everyone. 

The solution for me is that of camposmichel : 

> Hello everyone, I solved using the versions
> **"@ ionic-native / google-plus": "4.7.0" "cordova-plugin-googleplus": "5.2.1"**
> And adding the SHA1 debug in my Firebase application:
> **keytool -exportcert -alias androiddebugkey -keystore "C: \ Users \ Michel Fields \ .android \ debug.keystore" -list -v**
> **The default password is "android".**
> 
> It worked for me.

**Comment by EddyVerbruggen** [comment]
@restivof If you feel anything should be clarified better in the readme, then plz don't forget to open a PR.

**Comment by bhutada1992** [comment]
hii @SeaMeow please give steps to set google plus login

**Comment by bhutada1992** [comment]
@EddyVerbruggen  tried all way but no luck

**Comment by Dan890** [comment]
@bhutada1992  I still can't fix the problem,
my firebase app included the keys of debug & production. and also google services.json
i get error 16, when i login from debug\release anvironmet.

any ideas?

**Comment by bhutada1992** [comment]
@Dan890  nope tried all way but still geeting error 16

**Comment by JulienSopheos** [comment]
Same here, was working fine until one week ago ... Then getting error **16**, tried all suggestions mentionned here but none worked for me.

**Comment by jugg3rn4u7** [comment]
I've finally managed to fix this. There were multiple things that were off. So, here are the things I had to change -

1. Web app client id and reverse client id -
    - In the cloud platform dashboard, took the WEB APPLICATION client id (since its a hybrid app i work with) and used it, in the code.
    - Used reverse client id from the same dashboard and used the android's client id here. This somehow worked for ios as well which means that the reverse client id gets ignored and WEB APP Client id was used internally by the plugin :/

2. Changed build.json
    - This is placed in dist/mobile folder. If this file exists then for debug mode, the debug section gets used and for release mode, release section gets used. 
    - For my project, both were the same config pointing to release keystore, which probably caused error 16. But it was working before, so they must have fixed a bug on their end.
 
Here's my build.json -
{
    "android": {
        "debug": {
            "keystore": "~/.android/debug.keystore",
            "storePassword": "android",
            "alias": "androiddebugkey",
            "password" : "android",
            "keystoreType": ""
        },
        "release": {
            "keystore": "blah.keystore",
            "storePassword": "blah",
            "alias": "blah_key",
            "password" : "blah",
            "keystoreType": ""
        }
    }
}

If you're using Android studio to make builds, you need to make the same changes in debug-signing.properties file, which you'll find in dist/mobile/platforms/android

3. Make sure SHA matches.
   - Not sure why firebase has a relation with google login as firebase is related to push notification, but to be on safe side, add the SHA for BOTH release and debug keystores in the firebase console.

keytool -list -v -keystore blah.keystore (whichever release or debug, you can view SHA for that keystore file)

4. Make sure your backend is an authorized js origin.
   - After google authenticates, it redirects to YOUR backend.
   - Again on Google Cloud Platform, click on WEB APPLICATION CLIENT ID, to view the details.  
   - Check "Authorised JavaScript origins". Add list of valid origins (debug / release) for whatever environments.

I'm pretty sure, the issue was only concentration around build.json and keystores but I'm giving a full set of steps for the whole thing to work right. Let me know if you've found different set of steps to fix your problems.

**Comment by shuki770** [comment]
Hello everyone
I solved the problem by adding google app signing certificate to firebase settings

At Google Play Console -> Release management -> App signing
copy the app signing certificate SHA1 to firebase settings (app certificate fingerprint)

**Comment by Dan890** [comment]
I fix it,
I had to remove the webClientId properties from login function.
OLD - window.plugins.googleplus.login({ "webClientId": 'com.googleusercontent.apps.blabla },
New - window.plugins.googleplus.login({  },

Now it's working on Android debug & release.
need to check it again on iOS.

**Comment by bhutada1992** [comment]
I have solved problem in android at end.Thank you every one.

**Comment by SRV-Garg** [comment]
I have solved this issue by changing the SHA-1 fingerprint in firebase app. I think its only the SHA-1 fingerprint issue so try to list the fingerprint with the keytool command and change the latest one. Thank you

**Comment by mkimmet** [comment]
Same experience as @SRV-Garg - adding the SHA-1 worked for me.  In the firebase console under Project Settings for my Android app I added the SHA-1 there.  Which seemed a little counter intuitive to me since I used the Google Cloud Console to create the OAuth (Web) Client, and would think that settings like the SHA-1 would be needed there, rather than in the Firebase console.

**Comment by EddyVerbruggen** [comment]
https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/529#issuecomment-422058426

**Comment by goodspeed666** [comment]
Thank you @asoa, you save a man !

**Comment by kypergio** [comment]
Working for me. thanks a lot


> Hello everyone, I solved using the versions
> **"@ ionic-native / google-plus": "4.7.0" "cordova-plugin-googleplus": "5.2.1"**
> And adding the SHA1 debug in my Firebase application:
> **keytool -exportcert -alias androiddebugkey -keystore "C: \ Users \ Michel Fields \ .android \ debug.keystore" -list -v**
> **The default password is "android".**
> 
> It worked for me.

**Comment by bppearsoft** [comment]
If you are packaging with ionic pro and testing on an android device, you will need to test this feature using a release build of your app. I continuously got error code 16 with my development build despite trying many of the suggestions above but in the end it worked fine once I used a release build. Hopefully this might save someone else some time. Thanks for the plugin Eddy.

**Comment by asoap** [comment]
> Thank you @asoa, you save a man !
> 
> It's work for the development.
> But in the production(Google Play Store), I still got the error.
> May I know your fingerprint only one for the debug keystore ?
> 
> Thank you

I'm not sure how the google play store will work.  I think you need to find out how to set the signature for it.  As it no longer uses the android debug key.  But I haven't gotten that far yet.  So I can't say.

**Comment by bhutada1992** [comment]
user play store SHA1 key and add to firebase project

**Comment by decodeJarvis** [comment]
I my case
I get my SHA1 key with
  **keytool -exportcert -keystore ~/.android/debug.keystore -list -v**
and added to firebase project setting
and it worked!
Enjoya

**Comment by chamupathiBlimpIT** [comment]
any updates in this? still getting error 16

**Comment by decodeJarvis** [comment]
> any updates in this? still getting error 16

you should have same SHA1 key in firebase and local machine
for both debug build and production build

**Comment by chamupathiBlimpIT** [comment]
> > any updates in this? still getting error 16
> 
> you should have same SHA1 key in firebase and local machine
> for both debug build and production build

yes I use 
the same SHA1 key in local machine, firebase 
using reversed client id as: com.googleusercontent.apps.XXXXXX
using web application client id -> web client -> client id (auto generated ) : XXXX..apps.googleusercontent.com

**Comment by decodeJarvis** [comment]
thats right, but you also need to add your sha-1 or sha-256 fingerprint of debug key and production key which looks like this,

**d6:ef:21:45:86:10:b9:ff:92:3g:95:10:1e:d4:g2:85:30:9a:e2:60**

which we can get via this command 
**keytool -exportcert -keystore path/to/your/key_name.keystore -list -v**

and you have to add it in 
firebase project setting> general > * your app name(android) >  Add fingerprint

hope this works
or this might help 
**https://medium.freecodecamp.org/how-to-integrate-google-login-into-an-ionic-app-with-firebase-41cb69234919?gi=e20551dc9f1e**
-----
https://www.djamware.com/post/59094a2280aca7414e78a63d/ionic-3-google-plus-authentication-tutorial

**Comment by mesqueeb** [comment]
@decodeJarvis 
I still have error 16 after your advice.
Would you be so cool to see if you have any extra advice for me?
: D

1. I opened cordova's android folder in Android Studio
2. I created a key-store and signed PKG via Android Studio
3. I sign up for Google Play console and submit my signed PKG as internal test
4. I receive my **App signing certificate** (& fingerprints) and **Upload certificate** (& fingerprints) via the Google Play console. They look like this:
<img width="315" alt="2019-01-05 20 57 03" src="https://user-images.githubusercontent.com/3253920/50724772-fbfb4c80-1135-11e9-9713-24e37f9cfc2f.png">
5. I go to Firebase console
6. I add both the **App signing** and **Upload** SHA1 fingerprints at `project settings > general > your app name(android) > Add fingerprint` 
7. I double check on Google Cloud console at Credentials > OAuth 2.0 client IDs and it has: **Android client for com.<myid> (auto created by Google Service)** two times, once for each fingerprint. They both have new client id's
8. Since I am building a hybrid app, I am using the original firebase **Web client** (auto created by Google Service) client ID and manually reversed that for the reverse ID for this plugin.
9. package.json and config.xml are set up properly
10. I have set up a `build.json` file with the information of my keystore like in the [cordova docs](https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html#signing-an-app).
11. when running `cordova run android` it notifies me that it has found the build.json and reads my keystore.

### Yet in spite all this I still get error `16`

Could it be because the client id in package.json is set to the web-client id and not the android one?
```json
"cordova": {
    "plugins": {
        "cordova-plugin-whitelist": {},
        "cordova-plugin-googleplus": {
            "REVERSED_CLIENT_ID": "com.googleusercontent.apps.1062218916578-oc5pu1atcblj7fcpq571uded5v9u3vof",
            "WEB_APPLICATION_CLIENT_ID": "1062218916578-oc5pu1atcblj7fcpq571uded5v9u3vof.apps.googleusercontent.com",
            "PLAY_SERVICES_VERSION": "+"
        }
    },
```

**Comment by mesqueeb** [comment]
@jugg3rn4u7 
Your comment also stood out for me. Especially when you say:

> For my project, both were the same config pointing to release keystore, which probably caused error 16. But it was working before, so they must have fixed a bug on their end.

I don't have a separate debug key, just one local keystore with one key which I called "upload-key". Google Play Console shows me an `App signing certificate (& fingerprint)` and `Upload certificate (& fingerprint)` on the other hand. These are the ones I added to firebase console.

But now when it comes to the build file you mention. Since I only have 1 local key, I gave the same exact keystore details to `debug` and `release`. You mean to say that this can cause the problem?

What am I to do if I do not have a separate debug key?

**Comment by mesqueeb** [comment]
@bppearsoft I am similar to your case. I have tried all the solutions above but could not solve the error 16.

You told me to try with a "release build of your app". Do you have any information how to do this? I'm not sure what you mean since when you do `cordova run` it will create a APK and sign it any way right? And is this APK not the same as "a release build"?

Any information on how to test with a "release build" is much appreciated.

**Comment by decodeJarvis** [comment]
> 16

does you config.xml have this 

**
<plugin name="cordova-plugin-googleplus" spec="^5.3.2">
        <variable name="REVERSED_CLIENT_ID" value="com.googleusercontent.apps.123123.asdasd" />
        <variable name="WEB_APPLICATION_CLIENT_ID" value="123123-asdasd.apps.googleusercontent.com" />
        <variable name="PLAY_SERVICES_VERSION" value="11.8.0" />
 </plugin>
**

 if not then try removing it and adding in this way,
ionic cordova plugin add cordova-plugin-googleplus --save --variable REVERSED_CLIENT_ID=com.googleusercontent.apps.123123.asdasd --variable WEB_APPLICATION_CLIENT_ID=123123-asdasd.apps.googleusercontent.com

and after adding plugin
remove platform and add it again to apply changes
hope it helps

**Comment by mesqueeb** [comment]
@decodeJarvis Thanks for your message.
Yes I just tried again now with my `Web client (auto created by Google Service)` client ID and that one in reverse:
1. `cordova plugin remove cordova-plugin-googleplus`
2. add the plugin again with that command you wrote and correct id's
3. `cordova platform rm android`
4. `cordova platform add android`
But still the same error.

**Comment by mesqueeb** [comment]
@decodeJarvis Thanks for your support.
The app builds and loads fine. I have a login button and do a:
```js
window.plugins.googleplus.login(
  {webClientId: clientIdOauth},
  result => {
    console.log('[cordova] login success', JSON.stringify(result, null, 2))
  },
  msg2 => {
    console.error('[cordova] login error', msg2)
  }
)
```
Here I get a modal and choose my gmail account, after which the modal closes and I get error 16.

**Comment by mesqueeb** [comment]
Maybe the app that is built and opened in the emulator by `cordova run` is not signed properly? How can i double check this?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by decodeJarvis** [comment]
> double

its signed properly,
have you added this ?
![screen shot 2019-01-05 at 10 29 23 pm](https://user-images.githubusercontent.com/20205821/50726920-914c1000-1139-11e9-8e9b-d8f1295f9163.png)

or else
i my case i added all  (upload certificate key, production key and debug key) three keys and both sha-1 and sha-256 in firebase then it worked

![screen shot 2019-01-05 at 10 32 48 pm](https://user-images.githubusercontent.com/20205821/50727085-9dd16800-113b-11e9-9204-1b1cdf3fe5cf.png)

hope this helps

**Comment by mesqueeb** [comment]
@decodeJarvis I finally able to not get error 16!!!!

### The solution
The error was because of the signatures.

Of course make sure that this build is signed by adding the correct keystore info to build.json as in my step 10 of [this post](https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/529#issuecomment-451658150) a little bit up in this thread. Those signatures were whitelisted in the Oauth settings so I didn't understand why it didn't work.

It turns out cordova automatically runs in a "debug mode" which uses completely different signatures.

1. to ensure you are running (testing) your public release in the emulator and correctly signed just do `cordova run android --release --emulator`
the release flag will make sure that that one is chosen.

now I was also able to make it work with authentication in in a debug environment.
When reading through the documentation again, my eye caught the section on [Sign your debug build](https://developer.android.com/studio/publish/app-signing#debug-mode). Where it states:

> When running or debugging your project from the IDE, Android Studio automatically signs your app with a debug certificate generated by the Android SDK tools. The first time you run or debug your project in Android Studio, the IDE automatically creates the debug keystore and certificate in $HOME/.android/debug.keystore, and sets the keystore and key passwords.

So i went looking if I had such a folder by doing in terminal:
```
ls /Users/myusername/.android
```
And there I saw this keystore called "debug.keystore" which I had never seen before! So I try to get the SHA1 signature fingerprint from this keystore by doing:
```
keytool -exportcert -keystore /Users/myusername/.android/debug.keystore  -list -v
```
and use password `android` which is the default password for that keystore!

2. Copy this completely new SHA1 fingerprint and add it to your Firebase console `SHA certificate fingerprints` list as well! Then your authentication will work even on the debug mode!!

**Comment by mesqueeb** [comment]
I add a part on this to the ReadMe. #563

**Comment by EddyVerbruggen** [comment]
@mesqueeb You're the best, thanks for adding this!

**Comment by chamupathiBlimpIT** [comment]
SOLVED!,

in my case, 

keytool 
`keytool -exportcert -keystore ~/.android/debug.keystore -list -v`
outputs an invalid SHA1 for some reason,

so I went to the apk output folder

`keytool -printcert -jarfile <my-app>.apk `

then it displays apps SHA1.

add it to firebase and google console.

**Comment by harshzalavadiya** [comment]
Suddenly after today afternoon application started giving error code 16.
and It's not SHA Issue because it's not working on previous builds of application as well where it used to work perfectly. plus i have tried creating fresh firebase and ionic application same 16

Can anyone confirm that it's working from their side possibly APAC region?

**Comment by tallcoder** [comment]
This was happening to me, even after making sure my keys and such were correct and pretty much everything else. I'm using Ionic 4. 

I got it working after upgrading my `cordova-plugin-ionic-webview` to `4.0.1`. I just wanted to post that in case someone else using Ionic 4 runs into this and can't get it working, even after getting the keys and such added correctly.

**Comment by babitakapoor112** [comment]
@camposmichel , I am getting error code 10 (DEVELOPER ERROR ) after implementation of debug keystore SHA1 in firebase console . I have downloaded the new google service.json file in the project.

**Comment by evinkuraga** [comment]
I have no debug.keystore. Is that normal? I don't know how relevant this is, but I have a build.json file that references an android.jks file. I'm assuming that's what is being used for the signing. 

Everything was working up until recently. I had to go in the project.properties file to change the play-service-auth and play-services-identity from version X to version +, but it would work after that.

I recently upgraded a few packages as well as the platform (8.0.0). Ever since, nothing seems to work. The docs seem to point at an app signing issue, but it's difficult to even debug this without having the debug.keystore file. I've attempted to extract it from the .jks file and uploaded the sha1 to both FB and google cloud, but still no success. Any advice? Thanks so much!
